tests/phpunit/CRM/Contribute/Form/ContributionTest.php

```diff
@@ -1324,6 +1324,52 @@ public function testCardTypeAndPanTruncation() {
     $this->assertEquals(CRM_Utils_Array::value('pan_truncation', $financialTrxn), 4567);
   }
 
+  /**
+   * Check payment processor is correctly assigned for a contribution page.
+   */
+  public function testContributionBasePreProcess() {
+    //Create contribution page with only pay later enabled.
+    $params = array(
+      'title' => "Test Contribution Page",
+      'financial_type_id' => 1,
+      'currency' => 'NZD',
+      'goal_amount' => 100,
+      'is_pay_later' => 1,
+      'is_monetary' => TRUE,
+      'is_active' => TRUE,
+      'is_email_receipt' => TRUE,
+      'receipt_from_email' => 'yourconscience@donate.com',
+      'receipt_from_name' => 'Ego Freud',
+    );
+
+    $page1 = $this->callAPISuccess("contribution_page", 'create', $params);
+
+    //Execute CRM_Contribute_Form_ContributionBase preProcess
+    //and check the assignment of payment processors
+    $form = new CRM_Contribute_Form_ContributionBase();
```

Comment:
maybe it would help to load the form 'Main' rather than it's parent? There is a helper 'getForm' which might set things that matter